test/unit/pipelined_compilation/pipelined_compilation_test.bzl

```diff
@@ -0,0 +1,125 @@
+"""Unittests for rust rules."""
```

Comment:
Thanks for the extensive coverage!
One more test I would like to see is that in the case of 
`rust_library_a` -> `proc_macro` -> `rust_library_b`, when compiling `rust_library_a` object or rmeta file, we do not pass the `rust_library_b` `.rmeta` or `.rlib` as an input (we shouldn't propagate inputs from `proc_macro` depenendencies).
I think https://github.com/bazelbuild/rules_rust/blob/main/test/unit/proc_macro/leaks_deps/proc_macro_does_not_leak_deps.bzl will be a fitting place for such a test, as it tests the same for `.rlib` and native dependencies.

Reply:
Done.